scripts/update/update_disease_ontology_terms.py

```diff
@@ -0,0 +1,288 @@
+#!/usr/bin/env python3
+
+import sys
+import argparse
+import MySQLdb
+import requests
+import configparser
+import re
+
+
+def dump_ontology_data(db_host: str, db_port: int, db_name: str, user: str, password: str) -> dict[str, dict]:
+    """
+    Method to fetch all the OMIM and Mondo ontology terms stored in G2P.
+
+    Args:
+        db_host (str): G2P database host name
+        db_port (int): G2P database port number
+        db_name (str): G2P database name
+        user (str): user with read access
+        password (str): password
+
+    Returns:
+        dict[str, dict]: dictionary with ontology accession and its corresponding data
+    """
+
+    ontology_records = {}
+
+    sql = """   SELECT o.accession, o.term, o.description
+                FROM ontology_term o
+                LEFT JOIN source s ON s.id = o.source_id
+                WHERE s.name = 'Mondo' OR s.name = 'OMIM'
+          """
+
+    db = MySQLdb.connect(host=db_host, port=db_port, user=user, passwd=password, db=db_name)
+    cursor = db.cursor()
+    cursor.execute(sql)
+    data = cursor.fetchall()
+    for row in data:
+        try:
+            ontology_records[row[0]]
+        except KeyError:
+            ontology_records[row[0]] = {
+                "term": row[1],
+                "description": row[2]
+            }
+        else:
+            print(f"WARNING: duplicated ontology term '{row[0]}'")
+
+    db.close()
+
+    return ontology_records
+
+
+def analyse_terms(ontology_records: dict[str, dict]) -> tuple[dict[str, dict], list[str]]:
+    """
+    Method to analyse which ontology terms have to be updated or removed.
+    Records to update are:
+        - ontology terms with accession equal to the term
+
+    Args:
+        ontology_records (dict[str, dict]): dictionary with all ontology terms
+
+    Returns:
+        dict[str, dict]: dictionary with ontology terms to be updated
+        list[str]: list of ontology IDs that are obsolete
+    """    
+    records_to_update = {}
+    records_to_delete = []
+
+    for accession in ontology_records:
+        # Checks if the term name is the same as the accession
+        # For this scenario we have to query the Mondo API to get the
+        # correct term
+        if accession == ontology_records[accession]["term"]:
```

Comment:
`if accession == ontology_records[accession]["term"]` would not be true for OMIM records

Reply:
As discussed, if the term is different from the accession then the ontology term doesn't an update which is the case for most IDs.